### ISSUE TITLE
add `any` operator

### ISSUE THREAD

**Issue by helmbold** [Kotlin/kotlinx.coroutines#2239]
To make Kotlin Flow more suitable for a wide range of tasks, I suggest to add an [`any` function like in Reactor](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#any-java.util.function.Predicate-).

## Description

Emit a single boolean true if any of the values of this Flux sequence match the predicate.

## Signature

```kotlin
fun <T> Flow<T>.any(predicate: (T) -> Boolean)): Flow<Boolean>
```

**Comment by elizarov** [comment]
You can use `first { ... }` terminal operator with Kotln Flow. I am not sure why would you need `any`. Can you give an example?

**Comment by helmbold** [comment]
Agreed. Since `Flow` doesn't have the concept of `Mono` and would terminate after emitting the first (and only) element adding `any` wouldn't make a big difference.

I just want to suggest the development of some more operators, since I was missing some operators I know from Reactor recently. So my approach is to go the the list of `Flux` operators, check if they already exist in Flow or were already filed as an issue, and then create a new issue. But maybe I generate to much duplicates (due slightly different naming) or operators that could easily replaced with existing Flow functions. I don't want to bother you, so if you don't think my approach is helpful, let me now.

**Comment by JakeWharton** [comment]
> Since `Flow` doesn't have the concept of Mono

A `Mono` in coroutines is just a `suspend fun`, an asynchronous function invocation that can only produce a value or exception.

**Comment by elizarov** [comment]
This approach is not helpful. We don't have a goal of having all the Reactor operators. Our goals are completely opposite. We want to have a small core set of operators that people can easily learn and use to write their code. So, we are adding operators only when there are strong supporting use-cases to have them.

**Comment by helmbold** [comment]
Ok, thanks for clarification.

**Comment by jfsanchez91** [comment]
You can implement your own extension functions: 
``` kotlin
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.count
import kotlinx.coroutines.flow.first

suspend fun <T> Flow<T>.all(predicate: suspend (T) -> Boolean): Boolean {
    return this.count { !predicate(it) } == 0
}

suspend fun <T> Flow<T>.any(predicate: suspend (T) -> Boolean): Boolean {
    try {
        this.first { predicate(it) }
    } catch (e: NoSuchElementException) {
        return false
    }
    return true
}

```

**Comment by jdreddaway** [comment]
[jfsanchez91](https://github.com/jfsanchez91) it would be better performance to use this.firstOrNull() and check for null instead of try/catching.
